folding-schemes/src/folding/nova/mod.rs

```diff
@@ -249,6 +250,187 @@ where
     pub cf_U_i: CommittedInstance<C2>,
 }
 
+impl<
+        C1: CurveGroup,
+        GC1: CurveVar<C1, CF2<C1>> + ToConstraintFieldGadget<CF2<C1>>,
+        C2: CurveGroup,
+        GC2: CurveVar<C2, CF2<C2>>,
+        FC: FCircuit<C1::ScalarField>,
+        CS1: CommitmentScheme<C1>,
+        CS2: CommitmentScheme<C2>,
+    > CanonicalSerialize for Nova<C1, GC1, C2, GC2, FC, CS1, CS2>
+{
+    fn serialize_with_mode<W: Write>(
+        &self,
+        mut writer: W,
+        compress: ark_serialize::Compress,
+    ) -> Result<(), ark_serialize::SerializationError> {
+        // we serialize non-deterministic parameters
+        self.cs_params.serialize_with_mode(&mut writer, compress)?;
+        self.cf_cs_params
+            .serialize_with_mode(&mut writer, compress)?;
+        self.i.serialize_with_mode(&mut writer, compress)?;
+        self.z_0.serialize_with_mode(&mut writer, compress)?;
+        self.z_i.serialize_with_mode(&mut writer, compress)?;
+        self.w_i.serialize_with_mode(&mut writer, compress)?;
+        self.u_i.serialize_with_mode(&mut writer, compress)?;
+        self.W_i.serialize_with_mode(&mut writer, compress)?;
+        self.U_i.serialize_with_mode(&mut writer, compress)?;
+        self.cf_W_i.serialize_with_mode(&mut writer, compress)?;
+        self.cf_U_i.serialize_with_mode(&mut writer, compress)?;
+        self.r1cs.serialize_with_mode(&mut writer, compress)?;
+        self.cf_r1cs.serialize_with_mode(&mut writer, compress)?;
+        self.poseidon_config
```

Comment:
We serialize `PoseidonParams` since we can't implement an external trait (`CanonicalDeserialize`) on an external type (`PoseidonParams`).

Reply:
Same as in https://github.com/privacy-scaling-explorations/sonobe/pull/107#discussion_r1627145921